stdlib/src/testing/testing.mojo

```diff
@@ -300,29 +308,64 @@ fn assert_almost_equal[
         raise _assert_error(err, __call_location())
 
 
-fn _assert_equal_error(
-    lhs: String, rhs: String, msg: String, loc: _SourceLocation
-) -> String:
-    var err = (
-        "`left == right` comparison failed:\n   left: "
-        + lhs
-        + "\n  right: "
-        + rhs
-    )
-    if msg:
-        err += "\n  reason: " + msg
-    return _assert_error(err, loc)
+@always_inline
+fn assert_is[
+    T: StringableIdentifiable
+](lhs: Reference[T, _, _], rhs: Reference[T, _, _], msg: String = "") raises:
```

Comment:
Thanks for this, these are a natural extension of the assertion functions! 🙂 

I agree with the suggestion that we hold off on defining these in terms of `Reference`, and we avoid printing the `UnsafePointer.address_of(<lhs/rhs>)` below.

The issue is that it gives a false impression about what `is` actually does. `is` is like a "deep identity" check. As the test added later on demonstrates, two `PythonObject` instances on the stack will have different _stack_ addresses (if printed in the error case below), but still be `is` equivalent because their interior pointer points to the same reference-counted object.

For that reason, I suggest we change this signature to `lhs: T, rhs: T, ..`.

(Not for this PR, but if we really wanted to get the "interior" addresses for the purpose of printing them, that is something that types conforming to `Identifiable` would need to expose explicitly. Hypothetically, not suggesting this, there could be `Identifiable.__is_address__(self) -> Int` that e.g. for PythonObject would return the reference counted pointer address.)